server/channels/store/sqlstore/channel_store.go

```diff
@@ -1190,6 +1190,38 @@ func (s SqlChannelStore) GetChannelsByUser(userId string, includeDeleted bool, l
 	return channels, nil
 }
 
+func (s SqlChannelStore) GetBotChannelsByUser(userId string, opts store.ChannelSearchOpts) (model.ChannelList, error) {
+	query := s.getQueryBuilder().
+		Select("C.*").
+		From("Channels as C, ChannelMembers as CM1, ChannelMembers as CM2, Bots as B").
+		Where(sq.And{
+			sq.Expr("C.Id = CM1.ChannelId"),
+			sq.Eq{"CM1.UserId": userId},
+			sq.Eq{"C.Type": model.ChannelTypeDirect},
+			sq.Expr("C.Id = CM2.ChannelId"),
+			sq.Expr("CM2.UserId = B.UserId"),
+			sq.Eq{"B.DeleteAt": 0},
+		}).
+		OrderBy("C.Id ASC")
+
```

Comment:
@agarciamontoro would love your feedback on this new query and for visibility on this change from the server platform team